clients/fides-js/src/lib/consent-value.ts

```diff
@@ -22,20 +30,17 @@ export const resolveLegacyConsentValue = (
 };
 
 export const resolveConsentValue = (
-  notice: PrivacyNotice,
-  context: ConsentContext
+  notice: PrivacyNoticeWithPreference,
+  context: ConsentContext,
+  cookie: FidesCookie
 ): boolean => {
-  if (notice.current_preference) {
-    return transformUserPreferenceToBoolean(notice.current_preference);
-  }
   if (notice.consent_mechanism === ConsentMechanism.NOTICE_ONLY) {
     return true;
   }
-  const gpcEnabled =
-    !!notice.has_gpc_flag && context.globalPrivacyControl === true;
-  if (gpcEnabled) {
-    return false;
+  const preferenceExistsInCookie = noticeHasConsentInCookie(notice, cookie);
+  // Note about GPC - consent has already applied to the cookie at this point, so we can trust preference there
+  if (preferenceExistsInCookie) {
+    return <boolean>cookie.consent[notice.notice_key];
```

Comment:
I think this should work
```suggestion
    return !!cookie.consent[notice.notice_key];
```